internal/retry/retry.go

```diff
@@ -79,3 +79,19 @@ type Doer interface {
 	// It should return an error that can be checked for retriability.
 	Do(ctx context.Context) error
 }
+
+// DoerFunc is a convenience implementation of Doer for a closure.
+type DoerFunc func(context.Context) error
+
+// Do implements Doer.
```

Comment:
The average comment if your linters require public APIs to be commented.
Makes you question their usefulness. :D